diff -ur Thief-pristine/FICS.C Thief 1.23 Release Source/FICS.C
--- Thief-pristine/FICS.C	2008-09-12 15:04:48.000000000 -0400
+++ Thief 1.23 Release Source/FICS.C	2010-03-19 01:00:03.000000000 -0400
@@ -7,6 +7,12 @@
     COLORREF nC ;
 	FARPROC  LpProc ;
 
+    if (cS [0] == '\n')
+        {
+        TELNET_Write ("\n") ;
+        cS++ ;
+        }
+
     if (cS [0] == '\\')
         {
         if (Telnet.bDisplayContinueLine)
@@ -193,7 +199,7 @@
 
     //if (TELNET_MatchExp (cS, FICS_STYLE_12_SET ) ||
 
-    if (strstr (cS, FICS_STYLE_12_SET))
+    if (strstr (cS, FICS_STYLE_12_SET) || strstr (cS, FICS_STYLE_12_SET1))
         {
         nC = Telnet.clrCurrent ;
         Telnet.clrCurrent = clrColor [CLR_TELNET_NORMAL] ;
@@ -256,6 +262,15 @@
         cS++ ;
         }
 
+   //
+   // New FICS server sends \r\n, so we need to strip the \n
+   //
+   if (cS [0] == '\n')
+       {
+       TELNET_Write ("\n") ;
+       cS++ ;
+       }
+
     if (cS [1] == 'i' && cS [2] == 'c' && cS [3] == 's' && cS [4] == '%' && cS [5] == ' ')
         {        
 //        cS += FICS_PROMPT_LENGTH ;
diff -ur Thief-pristine/FICS.H Thief 1.23 Release Source/FICS.H
--- Thief-pristine/FICS.H	2008-09-12 14:59:20.000000000 -0400
+++ Thief 1.23 Release Source/FICS.H	2010-03-19 01:02:14.000000000 -0400
@@ -15,6 +15,7 @@
 #define FICS_ENTER                  "Press return to enter the server as"
 #define FICS_LOGIN_GUEST            "Press return to enter the FICS as"
 #define FICS_STYLE_12_SET           "Style 12 set."
+#define FICS_STYLE_12_SET1          "style set to 12."
 #define FICS_G1_INIT                "<g1> "
 #define FICS_12_INIT                "<12> "
 #define FICS_B1_INIT                "<b1> "
diff -ur Thief-pristine/GLOBAL.H Thief 1.23 Release Source/GLOBAL.H
--- Thief-pristine/GLOBAL.H	2008-09-22 20:59:26.000000000 -0400
+++ Thief 1.23 Release Source/GLOBAL.H	2010-03-19 00:54:31.000000000 -0400
@@ -29,7 +29,7 @@
 #ifdef  DEBUG
 #define THIEF_NICK                  "(The Beta, Debugging)"
 #else 
-#define THIEF_NICK                  "(The Release)"
+#define THIEF_NICK                  "(FatICS edition)"
 #endif
 
 #ifdef KICS
